src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2022,16 +2045,232 @@ private IOperation PeekOperand()
+                                    }
+
+                                    placeholderKeys.Add(argumentPlaceholder);
+                                    break;
+
+                                case InterpolatedStringArgumentPlaceholderKind.CallsiteArgument:
+                                    if (argumentPlaceholder.ArgumentIndex >= i)
+                                    {
+                                        // Out of position argument: just capture an invalid operation and reference it from the constructor
+                                        AddPlaceholder(argumentPlaceholder, new InvalidOperation(
+                                            ImmutableArray<IOperation>.Empty,
+                                            semanticModel: null,
+                                            syntax: argumentPlaceholder.Syntax,
+                                            type: argumentPlaceholder.Type,
+                                            constantValue: argumentPlaceholder.GetConstantValue(),
+                                            isImplicit: true));
+                                    }
+                                    else
+                                    {
+                                        Debug.Assert(visitedArguments[argumentPlaceholder.ArgumentIndex] is IFlowCaptureReferenceOperation or IDiscardOperation);
+                                        AddPlaceholder(argumentPlaceholder, visitedArguments[argumentPlaceholder.ArgumentIndex]);
+                                    }
+                                    placeholderKeys.Add(argumentPlaceholder);
+                                    break;
+                            }
+                        }
+                    }
+
+                    IOperation visitedArgument = VisitRequired(originalArgument);
+                    Debug.Assert(visitedArgument is IFlowCaptureReferenceOperation);
+                    visitedArguments.Add(visitedArgument);
+
+                    foreach (var key in placeholderKeys)
+                    {
+                        RemovePlaceholder(key);
+                    }
+
+                    placeholderKeys.Clear();
+
+                    var nextIndex = i + 1;
+                    if (nextIndex == arguments.Length || !hasInterpolatedStringHandlerArgument(arguments.AsSpan()[nextIndex..]))
+                    {
+                        i = nextIndex;
+                        return true;
+                    }
+
+                    return false;
+                }
+            }
+
+            static bool hasInterpolatedStringHandlerArgument(ReadOnlySpan<IArgumentOperation> arguments)
+            {
+                return arguments.Any(
+                    static arg => arg.Value switch
+                    {
+                        IInterpolatedStringHandlerCreationOperation { HandlerCreation: IObjectCreationOperation { Arguments: { } creationArgs } } =>
```

Comment:
>IInterpolatedStringHandlerCreationOperation [](http://example.com/codeflow?start=24&length=43)

I do not see any benefit from repeating this type check for every case within the switch. It feels like the code will be easier to follow when the type check is moved outside and performed once.